lib/lz4hc.c

```diff
@@ -736,24 +744,37 @@ LZ4_FORCE_INLINE int LZ4HC_compress_generic_internal (
 
     DEBUGLOG(4, "LZ4HC_compress_generic(%p, %p, %d)", ctx, src, *srcSizePtr);
 
-    if (limit == limitedDestSize && dstCapacity < 1) return 0;         /* Impossible to store anything */
-    if ((U32)*srcSizePtr > (U32)LZ4_MAX_INPUT_SIZE) return 0;          /* Unsupported input size (too large or negative) */
+    if (limit == limitedDestSize && dstCapacity < 1) {
```

Comment:
I know these two conditions produce errors, since we can't proceed. However, I believe the context is left in a clean state, since we haven't in fact performed any writes into the context yet. So I believe it's safe to not mark the context dirty here?

Reply:
This is a very good point which leads to a question whether here https://github.com/lz4/lz4/pull/580/files#diff-a4dc0fa3ccaea8a4219c81928cba271aR708 I have to revert my changes as well since no action was taken?

Reply:
Ah yes, I believe it should be safe to skip marking the context dirty there as well. Maybe with a note as to why they're not `goto _failure;`ing?